### DISCUSSION TITLE
Bug? Using provider with atom that derive from atomsWithStorage trigger twice in jotai v2

### DISCUSSION THREAD

**Discussion by devnomic** [pmndrs/jotai#1743] [General] [answered]
Using provider with atom that derive from atomsWithStorage trigger twice with initial value as first.

Minimal repro:
- Click the change id button.
- Click the toggle show button

jotai v2:
https://codesandbox.io/s/rough-dawn-45558b?file=/src/App.tsx

jotai v1: (same code, work as expected). 
https://codesandbox.io/s/headless-sun-bnxrvb?file=/src/App.tsx

![bbb](https://user-images.githubusercontent.com/101681592/216721452-b7e60123-3755-4bec-a5cf-fb73694fb69f.gif)

**Comment by dai-shi** [comment]
Yes, it's a design choice.
https://github.com/pmndrs/jotai/blob/c4ef100deaa86390cc2a8a475ad07efdbe11ad18/docs/guides/migrating-to-v2-api.mdx?plain=1#L213
(That one line is not very helpful though...)

See also: #1737

**Comment by devnomic** [comment, marked as answer]
I see,
But why it only happens when i wrap it with provider tag?
So what should i do if i want the same behavior like v1? 

My case i have a component that has a props that will initialize the atom. basically it's #826 so i need provider to wrap the component.

**Reply by dai-shi** [reply]
You are remounting the entire Provider. That's why.

If you want the sync behavior with localStorage, here's the simple, valid, and flexible solution:
https://github.com/pmndrs/jotai/blob/dae4418d330bb8d2ea151fc10ab5bcaaad595bd2/docs/guides/persistence.mdx?plain=1#L14-L22

https://codesandbox.io/s/currying-browser-0mps6p?file=/src/App.tsx

**Reply by devnomic** [reply]
Thank you!
